commander/src/bootstrapping/commands/generator/enable.ts

```diff
@@ -13,43 +13,94 @@
  *
  */
 
-import { BaseForgingCommand } from '../base_forging';
+import { Flags } from '@oclif/core';
+import { BaseGeneratorCommand } from '../base_generator';
 
-export abstract class EnableCommand extends BaseForgingCommand {
-	static description = 'Enable forging for given validator address.';
+const isLessThanZero = (value: number | undefined | null): boolean =>
+	value === null || value === undefined || value < 0;
+
+interface Status {
+	readonly address: string;
+	readonly height: number;
+	readonly maxHeightGenerated: number;
+	readonly maxHeightPrevoted: number;
+}
+
+export abstract class EnableCommand extends BaseGeneratorCommand {
+	static description = 'Enable block generation for given validator address.';
 
 	static examples = [
-		'generator:enable lsk24cd35u4jdq8szo3pnsqe5dsxwrnazyqqqg5eu 100 100 10',
-		'generator:enable lsk24cd35u4jdq8szo3pnsqe5dsxwrnazyqqqg5eu 100 100 10 --overwrite',
-		'generator:enable lsk24cd35u4jdq8szo3pnsqe5dsxwrnazyqqqg5eu 100 100 10 --data-path ./data',
-		'generator:enable lsk24cd35u4jdq8szo3pnsqe5dsxwrnazyqqqg5eu 100 100 10 --data-path ./data --password your_password',
+		'generator:enable lsk24cd35u4jdq8szo3pnsqe5dsxwrnazyqqqg5eu --use-status-value',
+		'generator:enable lsk24cd35u4jdq8szo3pnsqe5dsxwrnazyqqqg5eu --height=100 --max-height-generated=30 --max-height-prevoted=10',
+		'generator:enable lsk24cd35u4jdq8szo3pnsqe5dsxwrnazyqqqg5eu --height=100 --max-height-generated=30 --max-height-prevoted=10 --data-path ./data',
+		'generator:enable lsk24cd35u4jdq8szo3pnsqe5dsxwrnazyqqqg5eu --height=100 --max-height-generated=30 --max-height-prevoted=10 --data-path ./data --password your_password',
 	];
 
 	static flags = {
-		...BaseForgingCommand.flags,
+		...BaseGeneratorCommand.flags,
+		height: Flags.integer({
+			description: 'Last generated block height.',
+		}),
+		'max-height-generated': Flags.integer({
+			description: 'Validators largest previously generated height.',
+		}),
+		'max-height-prevoted': Flags.integer({
+			description: 'Validators largest prevoted height for a block.',
+		}),
+		'use-status-value': Flags.boolean({
+			description: 'use status value from the connected node',
```

Comment:
To nitpick, can we please change the message description to 'Use status value from the connected node.', just to have consistent pattern with other command descriptions?